src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -78,6 +82,35 @@ export class StudyViewPageStore {
 
     private _clinicalAttributesMetaSet: { [id: string]: ChartMeta } = {} as any;
 
+    @observable private _query: SingleGeneQuery[] = [];
+    @observable private _genes: Gene[] = [];
+    @observable geneQuery: string = '';
+
+    @bind
+    @action onCheckGene(hugoGeneSymbol: string) {
+        let geneSet= stringListToSet(this.geneQuery.trim().split(/\s+/g));
+        if (geneSet[hugoGeneSymbol]) {
```

Comment:
`geneSet[hugoGeneSymbol] = !geneSet[hugoGeneSymbol]`

Reply:
I don't think you're using the right data structures for the task here. What you want is a set of selected genes, then you can do 
```
@computed get geneQuery() { 
    return Object.keys(this._selectedGenes).filter(g=>!!this._selectedGenes[g]).join(' ');
}
@action set geneQuery(q:string) {
    this._selectedGenes = stringListToSet(q);
}
```